src/System.Private.ServiceModel/tests/Scenarios/SelfHostWcfService/IWcfDuplexService.cs

```diff
@@ -31,4 +32,17 @@ public interface IDuplexChannelCallback
         [OperationContract(IsOneWay = true)]
         void OnPingCallback(Guid guid);
     }
+
+    [ServiceContract(CallbackContract = typeof(IWcfDuplexTaskReturnCallback))]
+    public interface IWcfDuplexTaskReturnService
+    {
+        [OperationContract]
+        Task<Guid> Ping(Guid guid);
+    }
+
+    public interface IWcfDuplexTaskReturnCallback
+    {
+        [OperationContract]
+        Task<Guid> ServicePingCallback(Guid guid);
```

Comment:
I could but then I would have to put that Guid back into a Task in the service operation in order to return it back to the client. Seems easier/simpler this way?